visualizations/frontend/other/TPdfTable.vue

```diff
@@ -0,0 +1,149 @@
+      </table>
+    </div>
+    <slot
+      name="footer"
+      :rows-limit="rowsLimit"
+      :rows-total="rowsTotal"
+      :loading="loading"
+    >
+      <div class="pt-4 text-right">
+        Showing {{ rowsLimit }} of {{ rowsTotal }}
+      </div>
+    </slot>
+  </div>
+</template>
+
+<script>
+export default {
+  props: {
+    hiddenColumns: {
+      type: Array,
+      default: () => [],
+    },
+    dynamicColumns: {
+      type: Array,
+      default: () => [],
+    },
+    customColumns: {
+      type: Array,
+      default: () => [],
+    },
+    rowsLimit: {
+      type: Number,
+      default: 5,
+    },
+    containerClass: {
+      type: String,
+      default: "",
+    },
+    columnsClass: {
+      type: String,
+      default: "",
+    },
+    rowsClass: {
+      type: String,
+      default: "",
+    },
+  },
+  data: () => ({
+    is_filter: true,
+    rowsTotal: 0,
+  }),
+  computed: {
+    columns() {
+      const columns = this.rows.length ? Object.keys(this.rows[0]) : [];
+      const filteredInternalColumns = columns.filter(
+        (col) => !this.hiddenColumns.includes(col)
+      );
+      return [...filteredInternalColumns, ...this.customColumns];
+    },
+    visibleLayerColumns() {
+      let columns = [];
+      for (let column of this.dynamicColumns) {
+        if (this.filteredColumns.includes(column.url_value)) {
+          columns.push(column.sql_value);
+        }
+      }
+      return columns;
+    },
+    filteredColumns() {
+      const layerColumnsFilterName = `${this.layer}_cols`;
+      const urlParam =
+        this.filters.filter((f) =>
+          f.name.includes(layerColumnsFilterName)
+        )[0] || {};
+      return urlParam.value ? urlParam.value.split("|") : [];
+    },
+  },
+  created() {
+    this.fetchOnMounted = false;
+  },
+  mounted() {
+    this.prepareFilters();
+    this.fetchLayerData();
+    this.fetchRowsCount();
+  },
+  methods: {
+    prepareFilters() {
+      this.setLayerFilter("limit", this.rowsLimit);
+      this.setLayerFilter("offset", 0); // Always showing first page on PDF, pagination not required there.
+      this.setLayerFilter("orderBy", this.getFilterState(`${this.layer}_sort`));
+      this.setLayerFilter(
+        "column_list",
+        JSON.stringify(this.visibleLayerColumns)
+      );
+    },
+    fetchRowsCount() {
+      this.$store
+        .dispatch("layers/fetchLayerCount", this.createRequestPayload())
+        .then((rowsTotal) => {
+          this.rowsTotal = rowsTotal;
+        });
+    },
+    createRequestPayload() {
+      return {
+        render: {
+          visualization: this.tag,
+        },
+        target: this.tag_unique,
+        layer: this.layer,
+      };
+    },
+    setLayerFilter(filterName, filterValue) {
+      const table = this.$store.state.layers.components[this.tag_unique];
+      if (!table.filters) {
+        table.filters = {};
+      }
+      window.Vue.set(table.filters, filterName, filterValue);
```

Comment:
This looks like a hack. We should add a new mutation to the layers' store to store filters. Similar to these ones: https://github.com/snyk/topcoat-core/blob/main/frontend/src/store/modules/layers.js#L397-L426

Reply:
Actually, I think that something like this should work:
```js
  this.storeAttribute({
    target: this.tag_unique,
    attribute: "filters",
    value: value,
  });
```

Where value should be a copy of the current filters plus the one you want to add:
[...currentFilter, {filterName: filterValue}]
We need to check the right shape of filters.

Reply:
@egcastro Yes, that looks much cleaner, pushed it, thanks for correction.